src/test/java/org/benetech/servicenet/adapter/laac/LAACCompleteDataAdapterTest.java

```diff
@@ -0,0 +1,173 @@
+package org.benetech.servicenet.adapter.laac;
+
+import org.benetech.servicenet.adapter.AdapterTestsUtils;
+import org.benetech.servicenet.adapter.shared.model.SingleImportData;
+import org.benetech.servicenet.domain.Contact;
+import org.benetech.servicenet.domain.DataImportReport;
+import org.benetech.servicenet.domain.Eligibility;
+import org.benetech.servicenet.domain.Language;
+import org.benetech.servicenet.domain.Location;
+import org.benetech.servicenet.domain.Organization;
+import org.benetech.servicenet.domain.Phone;
+import org.benetech.servicenet.domain.PhysicalAddress;
+import org.benetech.servicenet.domain.Service;
+import org.benetech.servicenet.service.ImportService;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class LAACCompleteDataAdapterTest {
+
+    private static final String COMPLETE_JSON = "laac/complete.json";
+    private static final String PROVIDER_NAME = "LAAC";
+
+    @InjectMocks
+    private LAACDataAdapter adapter;
+
+    @Mock
+    private ImportService importService;
+
+    @Before
+    public void setUp() throws IOException {
+        MockitoAnnotations.initMocks(this);
+
+        when(importService.createOrUpdateService(any(Service.class),
+            anyString(), anyString(), any(DataImportReport.class)))
+            .thenReturn(new Service());
+
+        when(importService.createOrUpdateLocation(any(Location.class), anyString(), anyString()))
+            .thenReturn(new Location());
+
+        when(importService.createOrUpdateOrganization(any(Organization.class), anyString(), anyString(),
+            any(DataImportReport.class)))
+            .thenReturn(new Organization());
+
+        String json = AdapterTestsUtils.readResourceAsString(COMPLETE_JSON);
+        SingleImportData importData = new SingleImportData(json, new DataImportReport(), PROVIDER_NAME, true);
+        adapter.importData(importData);
```

Comment:
This is not part of the setup process and should be included in each test method.